### PR TITLE
Add replica set type

### PR DESCRIPTION
**Pull request by priyawadhwa**
First part of fixing #250 is to add support for validating ReplicaSets to kritis

### CODE REVIEW

File: pkg/kritis/admission/admission.go [hunk 1]
```diff
@@ -91,6 +92,16 @@ func handlePod(ar *v1beta1.AdmissionReview, admitResponse *v1beta1.AdmissionRevi
 	return nil
 }
 
+func handleReplicaSet(ar *v1beta1.AdmissionReview, admitResponse *v1beta1.AdmissionReview) error {
+	glog.Info("handling replica set...")
```
Comment (tstromberg) [inline pkg/kritis/admission/admission.go]: As an operator, I don't think I would find this message useful unless it mentioned which replica set it is handling. I suggest either adding more context, or using glog.V(2).Info.

Comment (priyawadhwa) [reply pkg/kritis/admission/admission.go]: for sure, added more info to this error and the ones for deployment/pods

File: pkg/kritis/admission/admission.go [hunk 2]
```diff
@@ -225,6 +236,20 @@ func reviewPod(pod *v1.Pod, ar *v1beta1.AdmissionReview) {
 	reviewImages(pods.Images(*pod), pod.Namespace, pod, ar)
 }
 
+func reviewReplicaSet(replicaSet *appsv1.ReplicaSet, ar *v1beta1.AdmissionReview) {
+	// First, check for a breakglass annotation on the replica set
+	if checkBreakglass(&replicaSet.ObjectMeta) {
+		glog.Infof("found breakglass annotation for %s, returning successful status", replicaSet.Name)
+		return
+	}
+	for _, c := range replicaSet.Spec.Template.Spec.InitContainers {
```
Comment (tejal29) [inline pkg/kritis/admission/admission.go]: How about we collect all images together. That ways, the reviewImages will prune duplicate images.
Calling them one by one would not do that.

Comment (tejal29) [reply pkg/kritis/admission/admission.go]: Can we add unit tests for this?

Comment (priyawadhwa) [reply pkg/kritis/admission/admission.go]: Done!

File: pkg/kritis/admission/images.go [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+	for _, ic := range pod.Spec.InitContainers {
+		images = append(images, ic.Image)
+	}
+	for _, c := range pod.Spec.Containers {
+		images = append(images, c.Image)
+	}
+	return images
+}
+
+// DeploymentImages returns a list of images in a pod
```
Comment (tstromberg) [inline pkg/kritis/admission/images.go]: This docstrings for these 3 functions should be different from one another.

Comment (priyawadhwa) [reply pkg/kritis/admission/images.go]: done

**Pull request merged**